func_sig_registry/registry/models.py

```diff
@@ -141,3 +150,99 @@ def get_hex_display(self):
 
     def get_bytes_display(self):
         return force_text(self.bytes4_signature)
+
+
+class EventSignature(models.Model):
+    created_at = models.DateTimeField(auto_now_add=True, db_index=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    text_signature = models.TextField(unique=True,
+                                        validators=[
+                                            MinLengthValidator(3), MaxLengthValidator(4096)
+                                        ])
+
+    bytes_signature = models.BinaryField(max_length=32,
+                                            unique=False,
+                                            validators=[MinLengthValidator(32)])
+                                          
+    hex_signature = models.CharField(max_length=64,
+                                        unique=False,
+                                        validators=[MinLengthValidator(64)])
+
+    def __str__(self):
+        return self.text_signature
+
+    def clean_fields(self, exclude=None):
+        try:
+            #TODO
+            self.text_signature = normalize_event_signature(self.text_signature)
+        except ValueError:
+            raise ValidationError('Unknown signature format')
+
+    class Meta:
+        unique_together = (
+            ('text_signature', 'bytes_signature'),
```

Comment:
I don't understand the need for this uniqueness constraint...  Since we expect both of these values to be individually unique.  I think this constraint would end up being satisfied implicitly by our data set but is not actually necessary.